keras/backend/tensorflow_backend.py

```diff
@@ -806,7 +783,16 @@ def random_normal_variable(shape, mean, scale, dtype=None,
     """
     if dtype is None:
         dtype = floatx()
-    tf_dtype = _convert_string_dtype(dtype)
+
+    supported_types = {
+        'float16', 'float32', 'float64',
+        'int16', 'int32', 'int64',
+        'uint8', 'uint16'}
+    if dtype not in supported_types:
+        raise ValueError('dtype only supports {},'
+                         'while get {}'.format(supported_types, dtype))
```

Comment:
You can remove this check. The error will be raised downstream anyway.

Reply:
However, its test case seems to require the check. see [code here](https://github.com/fchollet/keras/blob/19862b0b1d52e7636b30b0a0955ae4ecf2544edb/tests/keras/backend/backend_test.py#L222).

```python
# not supported dtype
for dtype in ['int16', 'int32', 'int64', 'uint8', 'uint16', 'double']:
    with pytest.raises(ValueError):
        ztf = KTF.random_normal_variable((2, 3), 0, 1, dtype=dtype)
```

Reply:
Then remove the test case. I assume what's being raised is a TypeError and that's why it would fail. It doesn't make sense to have this check for this function and nowhere else.

Reply:
OK, removed.